### ISSUE TITLE
Reply/ref number system is confusing

### ISSUE THREAD

**Issue by maiamcc** [billygoatsgruff/ccrb-bot#2]
I took this for a spin and found the "multiple matches, reply with ref number" system confusing, especially b/c the ref I replied with didn't always work, the options for the cop I meant came in multiple different tweets, both were replies to the FIRST tweet instead of replies to each other so I couldn't view the whole thread easily...

I propose that if it gets an ambiguous query, the bot makes you query again with more details.

E.g.
1. user tweets name, unambiguous:
    * user: @bot Frank Unique Name the Second
    * bot: <gives info>
2. user tweets badge, unambiguous
    * user: @bot 123789
    * bot: <gives info>
3. user tweets name, ambiguous
    * user: @bot Smith
    * bot: Sorry, you could be referring to multiple people. Try querying again by full name or badge number. Did you mean: John Smith, badge 123? Jane Smith, badge 456? <etc.>

Basically trying to avoid a chain of replies, and the bot having to keep conversations in memory, by giving the user all the info they need to make a second, clean request.

**Comment by maiamcc** [comment]
Rats, I had hoped badge numbers would be unique and unambiguous, but sounds like A. they're not guaranteed to be unique, and B. we also need to support fuzzy matching (e.g. for the case where there's a blurred photo and only a few badge numbers are visible). But this complicates things: how do we distinguish between a user who's searching 123 as a fuzzy match vs. someone who knows they want exactly badge 123 and _not_ badge 1234?

Thing one is that I think we should allow searching by name and badge together, since there might be lots of John Smith's and lots of badge 1234's but only one John Smith with badge 1234 (...god willing. If that's not the case, then we have no unique identifier ugh.) So someone could search for `John Smith 1234` and we'd parse that as a name and then a badge no.

Thing two, open question: do we let users specify when they want fuzzy and non-fuzzy matching? Maybe `123` for exact match and `123*` for fuzzy?

I'm imagining the flow
- user: @ bot 123
- bot: Sorry, you could be referring to multiple people. Try querying again by full name or badge number. Did you mean: John Smith, badge 123? Jane Doe, badge 1234? <etc.>
- user @ bot John Smith 123

The intermediary step, if you already know their full badge number, is kind of annoying, but not the worst, IF the list of options returned is manageable. If it's annoying large, it'll make the app borderline unusable. (Which raises a second question, what do we do with really really long lists of options... ugh.)

**Comment by mNov** [comment]
re: really long lists of options, could the bot just list the first N best matches and ask the user if they want to see more options? "best" meaning as close to an exact match as possible - probably in this example John Smith 123 is the better match than Jane Doe 1234 since the badge number exactly matches for 123.

**Comment by maiamcc** [comment]
> re: really long lists of options, could the bot just list the first N best matches and ask the user if they want to see more options? "best" meaning as close to an exact match as possible - probably in this example John Smith 123 is the better match than Jane Doe 1234 since the badge number exactly matches for 123.

FYI I'm forking this conversation over to #5 because it's a slightly different question than what the query flow should be like, but this is a good idea! Replied over in the other issue.
